File: dataload/linker/src/main/java/ExtractIriFromPropertyName.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class ExtractIriFromPropertyName {

    /* We have some added (by owl2json) properties like
    		relatedTo+http://....
		negativePropertyAssertion+http://...
	which consist of some alphabetic prefix, a + character, and the IRI
	this extracts the IRIs from them
	otherwise just returns the property name
	*/
    private static final Pattern namePattern = Pattern.compile( "^([A-z]+)\\+(.+)$" );
    public static String extract(String propertyName) {
        Matcher m = namePattern.matcher(propertyName);

        if(m.find()) {
            return m.group(2);
        } else {
            return propertyName;
        }
    }

}

File: dataload/linker/link/src/extract_iri_from_property_name.rs
use regex::Regex;
use std::sync::LazyLock;

/// Extract IRI from property names that have prefixes like "relatedTo+http://..."
/// or "negativePropertyAssertion+http://..."
///
/// If the property name matches the pattern "^[A-z]+\+(.+)$", returns the IRI part.
/// Otherwise returns the property name unchanged.

static NAME_PATTERN: LazyLock<Regex> = LazyLock::new(|| {
    Regex::new(r"^[A-Za-z]+\+(.+)$").unwrap()
});

pub fn extract(property_name: &str) -> &str {
    if let Some(captures) = NAME_PATTERN.captures(property_name) {
        if let Some(m) = captures.get(1) {
            // Return the captured group - we need to return a reference to the input
            // The capture matches a slice of the input string
            return m.as_str();
        }
    }
    property_name
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_extract_simple() {
        assert_eq!(extract("label"), "label");
        assert_eq!(extract("http://example.org/foo"), "http://example.org/foo");
    }

    #[test]
    fn test_extract_with_prefix() {
        assert_eq!(extract("relatedTo+http://example.org/foo"), "http://example.org/foo");
        assert_eq!(extract("negativePropertyAssertion+http://example.org/bar"), "http://example.org/bar");
    }
}
